test/trace_api_test.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/trace.h>
+
+#include "testutil.h"
+
+static int test_trace_categories(void)
+{
+    int cat_num;
+
+    for (cat_num = -1; cat_num <= OSSL_TRACE_CATEGORY_NUM + 1; ++cat_num) {
+        const char *cat_name = OSSL_trace_get_category_name(cat_num);
+        int is_cat_name_eq = 0;
+        int ret_cat_num;
+        int expected_ret;
+
+        switch (cat_num) {
+#define CASE(name) \
+        case OSSL_TRACE_CATEGORY_##name: \
+            is_cat_name_eq = TEST_str_eq(cat_name, #name); \
+            break;
```

Comment:
A recommendation: drop the final `;`.  That allows using using the macro in a way that source code formaters can handle a little better (yup, the `;` makes a difference in that sense):

``` C
            CASE(TRACE);
```